pkg/daemon/ceph/osd/volume.go

```diff
@@ -237,6 +237,12 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 		if device.Data == -1 {
 			logger.Infof("configuring new device %s", name)
 			deviceArg := path.Join("/dev", name)
+			// ceph-volume prefers to use /dev/mapper/<name> if the device has this kind of alias
+			for _, devlink := range device.PersistentDevicePaths {
+				if strings.HasPrefix(devlink, "/dev/mapper") {
+					deviceArg = devlink
```

Comment:
Could there be multiple devlinks that match? Should web be appending to a list instead of overwriting the var?

Reply:
In my Ubuntu 18.04, only one link will be created with the name of the contents of `/sys/block/<KNAME>/dm/name`.
This rule is stored in `/lib/udev/rules.d/55-dm.rules` and it is provided by Red Hat.
I think this is a general rule file.